### PR TITLE
initial work on splitting user guide and developer docs

### PR DESCRIPTION
**Pull request by indirectlylit**
# Details

### Summary

WIP

### Reviewer guidance

description of how to test the changes

### References

re #2596

# Contributor Checklist

- [ ] PR has the correct target milestone
- [ ] PR has the appropriate labels
- [ ] Changes in the PR do not introduce accessibility regressions ([confimed by testing with one of the recommended tools](http://kolibri.readthedocs.io/en/develop/dev/manual_testing.html#accessibility-a11y-testing)) 
- [ ] If PR is ready for review, it has been assigned or requests review from someone
- [ ] Documentation is updated as necessary
- [ ] External dependency files are updated (`yarn` and `pip`)
- [ ] If internal dependency is updated, link to diff is included
- [ ] Screenshots of any front-end changes are in the PR description
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] You've added yourself to AUTHORS.rst if you're not there

# Reviewer Checklist

- [x] Automated test coverage is satisfactory
- [x] PR has been fully tested manually

### CODE REVIEW

File: docs/_static/theme_overrides.css [hunk 1]
```diff
@@ -1,15 +0,0 @@
-
```
Comment (radinamatic) [inline docs/_static/theme_overrides.css]: This is actually useful for correctly displaying overflow in the table cells, any reason why it's deleted?

Comment (benjaoming) [reply docs/_static/theme_overrides.css]: It had issues building on RTD servers (but worked locally) so got commented out.. found a way to add it in #2599

**Comment by benjaoming** [comment]
@indirectlylit I see this as some kind of tech/refactor debt that is being paid back quite suddenly and without prior notice.. it requires changes to workflows on CrowdIn and RTD -- and then there's all the warnings that we just keep hauling around instead of fixing.

If the end-goal is just to save to save a poor user from a menu item called "Developer Docs", I don't see the priority being quite right... we have other debts that we diligently write down for later :)

I'd rather see us working on the contents and all those warnings instead of another breaking change in the structure, so please let's discuss this?

Also these disadvantages :

 * There's a number of greyzones where developers, advanced users and normal users overlap. These would require cross-referencing.
 * Versioning becomes harder and un-aligned between developer docs and user docs -- you suggest not to build developer versions of user docs, but why not? They are for all the pre-releases and to verify how they look etc.
 * Breaking up the docs means we have have to cross-reference via full URLs and not RST `:ref:` labels
 * Non-compatible with changes in previous versions of docs

**Comment by benjaoming** [comment]
On another minor note, it would make it easier to fix the most important of the 163 docs warnings by having some of them moved to a separate user-docs.

What about this as an option? Make a separate structure that builds only the user-facing docs as a smaller subset (remix) of the `docs/` folder? Meaning that the `Makefile` in `docs/user/Makefile` will build only that folder, symlinking `conf.py` and other global resources to `docs/user/`?

**Comment by indirectlylit** [comment]
Hey - sorry this came as a surprise. It's been discussed in the [10/16 dev meeting](https://docs.google.com/document/d/1qZKFhNwEZwvNmMQIcrfPRDpGNlEZiPxsY3UOb2gf3bE/edit#heading=h.ph9o5jgd0qt6) and some recent [documentation meetings](https://docs.google.com/document/d/18-Pi78DD20CazDbu6heJUrI1ndwhCOYaHmDYnq-a_S4/edit#heading=h.p11kt6uw9kot).

At a high level, the motivations were:

1. Developers by default want to see docs for the `develop` branch, whereas end-users by default want to see the docs for the latest stable release.
2. We risk disconcerting typical end-users (coaches and beginner admins) by exposing them to overly-technical documentation unless they have explicitly requested it.

To your specific points:

> There's a number of greyzones where developers, advanced users and normal users overlap. These would require cross-referencing. ... Breaking up the docs means we have have to cross-reference via full URLs and not RST :ref: labels

I just audited, and there are no cross-references between user and developer docs. It should be easy to continue writing documentation that avoids this issue, but if we really need to it looks like [there are options](http://www.sphinx-doc.org/en/stable/ext/intersphinx.html) besides full URLs.

> Versioning becomes harder and un-aligned between developer docs and user docs -- you suggest not to build developer versions of user docs, but why not? They are for all the pre-releases and to verify how they look etc.

The most important thing is that _default_ documentation for end-users is the latest stable. It's fine building and showing beta and alpha docs for early review, especially since the community will likely help with beta/alpha testing. The main risk is exposing novice users to incomplete docs for functionality that hasn't yet been released, which would be confusing.

> Non-compatible with changes in previous versions of docs

Could you elaborate? I was wondering if there might be implications here but couldn't think of any specific consequences.

**Comment by indirectlylit** [comment]
> What about this as an option? Make a separate structure that builds only the user-facing docs as a smaller subset (remix) of the docs/ folder? Meaning that the Makefile in docs/user/Makefile will build only that folder, symlinking conf.py and other global resources to docs/user/?

That sounds plausible. I'm trying to understand the pros/cons between this and breaking it into two projects, which (maybe naively) seemed simpler to me.

One general risk of remixing like this is an editor might write broken copy and links because they're only testing in one context, e.g.:

```
"See the `developer_docs` section of these docs for the gory details"
```

Would be broken and non-sensical.

**Comment by benjaoming** [comment]
> That sounds plausible. I'm trying to understand the pros/cons between this and breaking it into two projects, which (maybe naively) seemed simpler to me.

I was thinking about moving files around as little as possible -- and being able to share resources between the builds.

This is also re: my slight over-concern:

> > Non-compatible with changes in previous versions of docs
>
> Could you elaborate? I was wondering if there might be implications here but couldn't think of any specific consequences.

...the concern I had was that the refactoring would make it difficult to improve docs on current versions because files are moved around in `develop`.

> One general risk of remixing like this is an editor might write broken copy and links because they're only testing in one context, e.g.:
> 
> ```
> See the `developer_docs` section of these docs for the gory details.
> ```
> 
> Would be broken and non-sensical.

Yes, agreed -- We can choose whether to make developer docs a subset of user docs or the other way around. We would be able to refer into the subset, but not the other way back. The intention sounds like we want to be able to build user docs that have no reference to developer docs, anyways.

But I can also see the point in two completely separate projects, at least where the final output on `kolibri.readthedocs.io` and `kolibri-dev.readthedocs.io` are individual projects and don't have any identical pages... DRY-like

> I just audited, and there are no cross-references between user and developer docs. It should be easy to continue writing documentation that avoids this issue, but if we really need to it looks like there are options besides full URLs.

Regarding documentation readers/users, I think I've mentioned this before: But there will be more technical users, people implementing deployments. They are firm with the CLI, they are able to fix some bugs etc.. it's that greyzone. If we have user-only docs on `kolibri.readthedocs.io`, we would almost certainly have to add a link "For technical details, refer to our developer docs" somewhere, no?

Nice with the intersphinx mappings!! Good find!

And thanks for the links to the other meetings - no I wasn't aware of this :)

**Comment by indirectlylit** [comment]
> They are firm with the CLI, they are able to fix some bugs etc.. it's that greyzone. ... we would almost certainly have to add a link "For technical details, refer to our developer docs" somewhere, no?

I honestly don't know. Of course we want to encourage technically competent people to contribute bug fixes, and we could certainly link to the top-level of the developer docs from a 'contributing' section of the user docs.

However I can't imagine a concrete scenario where it would be desirable to include deep links into pages in the developer docs from other places in the user docs.

**Comment by benjaoming** [comment]
@indirectlylit merge conflicts + Tox config needs updating. But as for the idea itself, I think it's fine. Better to get it merged in quickly, than to let linger.

**Comment by indirectlylit** [comment]
should be fixed up now.

I ended up leaving the user docs at `/docs` and put the developer docs at /docs-developer` so that `kolibri.readthedocs.org` can maintain continuity with versions prior to 0.7

**Comment by indirectlylit** [comment]
> Not sure how the CrowdIn setup works for this, or what changes cold be needed in CrowdIn to reflect changes (possibly moving existing translations if there are any)?

Yes I'm sure some work needs to be done here. Other outstanding issues:

* cross-link to top-level urls
* fix up TOC of developer docs

I'll make issues for these this week

**Comment by benjaoming** [comment]
@indirectlylit 

One more issue: User Guide needs a better start page?

But I like the state of this PR, it should be merged so we can work on additional issues separately.

I can merge this now, just give me the signal and I'll get to work on any subsequent urgent fixes and the 3 issues that you create, just assign me!

**Comment by indirectlylit** [comment]
thank @benjaoming much appreciated!

**Comment by indirectlylit** [comment]
new follow-up issues:

* https://github.com/learningequality/kolibri/issues/2658
* https://github.com/learningequality/kolibri/issues/2659
* https://github.com/learningequality/kolibri/issues/2660
* https://github.com/learningequality/kolibri/issues/2661
* https://github.com/learningequality/kolibri/issues/2662

split between @radinamatic and @benjaoming

**Pull request merged**